tests/test_type_alias_type.py

```diff
@@ -0,0 +1,303 @@
+            'type': 'string_type',
+            'loc': ('dict[str,...]', 'a', 'str'),
+            'msg': 'Input should be a valid string',
+            'input': datetime.date(1992, 12, 11),
+        },
+        {
+            'type': 'int_type',
+            'loc': ('dict[str,...]', 'a', 'int'),
+            'msg': 'Input should be a valid integer',
+            'input': datetime.date(1992, 12, 11),
+        },
+        {
+            'type': 'float_type',
+            'loc': ('dict[str,...]', 'a', 'float'),
+            'msg': 'Input should be a valid number',
+            'input': datetime.date(1992, 12, 11),
+        },
+        {
+            'type': 'bool_type',
+            'loc': ('dict[str,...]', 'a', 'bool'),
+            'msg': 'Input should be a valid boolean',
+            'input': datetime.date(1992, 12, 11),
+        },
+        {
+            'type': 'string_type',
+            'loc': ('str',),
+            'msg': 'Input should be a valid string',
+            'input': {'a': datetime.date(1992, 12, 11)},
+        },
+        {
+            'type': 'int_type',
+            'loc': ('int',),
+            'msg': 'Input should be a valid integer',
+            'input': {'a': datetime.date(1992, 12, 11)},
+        },
+        {
+            'type': 'float_type',
+            'loc': ('float',),
+            'msg': 'Input should be a valid number',
+            'input': {'a': datetime.date(1992, 12, 11)},
+        },
+        {
+            'type': 'bool_type',
+            'loc': ('bool',),
+            'msg': 'Input should be a valid boolean',
+            'input': {'a': datetime.date(1992, 12, 11)},
+        },
+    ]
+
+    assert t.json_schema() == {
+        'allOf': [{'$ref': '#/$defs/JsonType'}],
+        '$defs': {
+            'JsonType': {
+                'anyOf': [
+                    {'type': 'array', 'items': {'$ref': '#/$defs/JsonType'}},
+                    {'type': 'object', 'additionalProperties': {'$ref': '#/$defs/JsonType'}},
+                    {'type': 'string'},
+                    {'type': 'integer'},
+                    {'type': 'number'},
+                    {'type': 'boolean'},
+                    {'type': 'null'},
+                ]
+            }
+        },
+    }
+
+
+def test_type_alias_annotated() -> None:
+    t = TypeAdapter(ShortMyList[int])
+
+    assert t.validate_python(['1']) == [1]
+
+    with pytest.raises(ValidationError) as exc_info:
+        t.validate_python([1, 2])
+    assert exc_info.value.errors(include_url=False) == [
+        {
+            'type': 'too_long',
+            'loc': (),
+            'msg': 'List should have at most 1 item after validation, not 2',
+            'input': [1, 2],
+            'ctx': {'field_type': 'List', 'max_length': 1, 'actual_length': 2},
+        }
+    ]
+
+    assert t.json_schema() == {'type': 'array', 'items': {'type': 'integer'}, 'maxItems': 1}
+
+
+@pytest.mark.xfail(reason='Not working yet, we cannot apply constraints to named/ref types')
```

Comment:
This is a tricky scenario:

```python
from typing_extensions import Annotated
from pydantic import BaseModel, PlainSerializer

def serialize_user(user: 'User'):
    return f'User(<{id(user)}>)'

class User(BaseModel):
    friends: Annotated['User', PlainSerializer(serialize_user)]
```

The problem is that at the point in time when we want to apply `PlainSerializer` to `User` the `User` schema is a `definition-ref` schema that is not yet defined. It so happens that you can add a `serialization` onto a `defintion-ref` schema, and in fact I can't think of any constraints you can add via `Annotated` that would need to modify the model schema itself (e.g. a `AfterValidator` would wrap it, only things like `Gt` modify the actual schema).

But that doesn't work in general: if the "named schema" is a `list[int]` (via `TypeAliasType`) then it should be possible to apply constraints like `MinLen`:

```T = TypeVar('T')
MyList = TypeAliasType('MyList', Annotated['MyList[T]', MinLen(1)], type_params=(T,))```

This should produce something along the lines of  `{'type': 'list', 'items_schema': {'type': 'definition-ref', 'schema_ref': 'MyList_int'}, 'min_length': 1, 'ref': 'MyList_int'}}`.

I tried quite hard to get this all working and could not, so I'm marking these as xfails for now and we can loop back and try to resolve them later.